specs/driver_spec.rb

```diff
@@ -77,4 +77,69 @@
       driver.average_rating.must_equal 0
     end
   end
+  describe "1.2 Aggregate Statistics" do
+    before do #use pry to get this info as hard coded data
+      dispatcher = RideShare::TripDispatcher.new
+      trip = dispatcher.trips.first
+      @driver = trip.driver
+      @duration = trip.duration
+    end
+
+    describe "Driver#total_revenue" do
+      it "returns total revenue" do
+        # use hard coded data to created expected output, not magic numbers
+        trips = @driver.trips
+        subtotal = 0
+         trips.each do |trip|
+          subtotal += (trip.cost - 1.65)
+        end
+        revenue_total = subtotal * 0.8
+        expected_value = revenue_total.round(2)
+
+        total = @driver.total_revenue
+        total.must_equal expected_value
+      end
+    end
+
+    describe "Driver#average_revenue" do
+      it "returns average revenue per hour spent driving" do
+
+        hour = 60 * 60
+        hour_time = 0
+        @driver.trips.each do |trip|
+          hour_time += (trip.duration / hour)
+        end
+        trip_count = @driver.trips.length
+        revenue = @driver.total_revenue
+        expected = (revenue / hour_time) / trip_count
+
+        test = @driver.average_revenue
+        test.must_equal expected
+
+      end
+    end
+  end
+  describe "interaction with TripDispatcher#request_trip" do
+    before do
+      @dispatcher = RideShare::TripDispatcher.new
```

Comment:
Good work testing these cases. I would argue that these tests should be grouped with the other test cases for `request_trip` in `trip_dispatcher_spec.rb`.